proof/refine/AARCH64/ArchInvsLemmas_H.thy

```diff
@@ -2191,1200 +165,104 @@ lemma cte_wp_at_cases':
-  by (fastforce simp: valid_refs'_def cteCaps_of_def elim!: ranE)
-
-lemma valid_cap_sizes_cteCaps:
-  "valid_cap_sizes' n (ctes_of s) = (\<forall>c \<in> ran (cteCaps_of s). 2 ^ capBits c \<le> n)"
-  apply (simp add: valid_cap_sizes'_def cteCaps_of_def)
-  apply (fastforce elim!: ranE)
-  done
-
-lemma cte_at_valid_cap_sizes_0:
-  "valid_cap_sizes' n ctes \<Longrightarrow> ctes p = Some cte \<Longrightarrow> 0 < n"
-  apply (clarsimp simp: valid_cap_sizes'_def)
-  apply (drule bspec, erule ranI)
-  apply (rule Suc_le_lessD, erule order_trans[rotated])
+                    is_aligned_mask[symmetric] objBits_simps'
+                    cte_level_bits_def magnitudeCheck_def
+                    return_def fail_def)
+   apply (clarsimp simp: bind_def return_def when_def fail_def
+                  split: option.splits)
+   apply simp
+  apply (erule(1) ps_clear_lookupAround2)
+    prefer 3
+    apply (simp add: loadObject_cte unless_def alignCheck_def
+                     is_aligned_mask[symmetric] objBits_simps'
+                     cte_level_bits_def magnitudeCheck_def
+                     return_def fail_def tcbCTableSlot_def tcbVTableSlot_def
+                     tcbIPCBufferSlot_def tcbReplySlot_def tcbCallerSlot_def
+              split: option.split_asm)
+     apply (clarsimp simp: bind_def tcb_cte_cases_def cteSizeBits_def split: if_split_asm)
+    apply (clarsimp simp: bind_def tcb_cte_cases_def iffD2[OF linorder_not_less]
+                          return_def cteSizeBits_def
+                   split: if_split_asm)
+   apply (subgoal_tac "p - n \<le> (p - n) + n", simp)
+   apply (erule is_aligned_no_wrap')
+    apply (simp add: word_bits_conv)
+   apply (simp add: tcb_cte_cases_def cteSizeBits_def split: if_split_asm)
+  apply (subgoal_tac "(p - n) + n \<le> (p - n) + 0x7FF")
+   apply (simp add: field_simps)
+  apply (rule word_plus_mono_right)
+   apply (simp add: tcb_cte_cases_def cteSizeBits_def split: if_split_asm)
+  apply (erule is_aligned_no_wrap')
   apply simp
   done
 
-lemma invs_valid_stateI' [elim!]:
-  "invs' s \<Longrightarrow> valid_state' s"
-  by (simp add: invs'_def)
-
-lemma tcb_at_invs' [elim!]:
-  "invs' s \<Longrightarrow> tcb_at' (ksCurThread s) s"
-  by (simp add: invs'_def cur_tcb'_def)
-
-lemma invs_valid_objs' [elim!]:
-  "invs' s \<Longrightarrow> valid_objs' s"
-  by (simp add: invs'_def valid_state'_def valid_pspace'_def)
-
-lemma invs_pspace_aligned' [elim!]:
-  "invs' s \<Longrightarrow> pspace_aligned' s"
-  by (simp add: invs'_def valid_state'_def valid_pspace'_def)
-
-lemma invs_pspace_distinct' [elim!]:
-  "invs' s \<Longrightarrow> pspace_distinct' s"
-  by (simp add: invs'_def valid_state'_def valid_pspace'_def)
-
-lemma invs_valid_pspace' [elim!]:
-  "invs' s \<Longrightarrow> valid_pspace' s"
-  by (simp add: invs'_def valid_state'_def)
+lemma cte_wp_atE'[consumes 1, case_names CTE TCB]:
+  assumes cte: "cte_wp_at' P ptr s"
```

Comment:
There is already a comment about sorting through this. My gut feeling is that if tcb_cte_cases can be kept generic, a lot of these can be generic.

Reply:
do not see a way of making them generic, looked through them again, nothing jumps out for migration, so removed the browsing FIXME